lambda/matching.ml

```diff
@@ -3323,13 +3323,47 @@ let check_partial pat_act_list =
 
 (* have toplevel handler when appropriate *)
 
-let check_total total lambda i handler_fun =
+type failer_kind =
+  | Raise_match_failure
+  | Reraise of lambda
+
+let failure_handler ~scopes loc ?(failer=Raise_match_failure) () =
```

Comment:
I don't think we're gaining much (or anything at all really) by having `failer` be an optional parameter instead of a labeled one.
Do you mind changing that?

Reply:
I pushed a commit that makes the change, but I find it a bit noisy.
(I have a commit "dans les cartons" to factorize the `check_total` logic, but I'm waiting for `Unused` clarifications from @maranget to finish and send it.)